### ISSUE TITLE
@type is not always redundant when using a type system

### ISSUE THREAD

**Issue by Lalem001** [gajus/eslint-plugin-jsdoc#1142]
## Expected behavior

Some types are generics and the default value for the generic variables, if they exist, may not be correct for the usage. An example would be setting the ExecutionContext in AVA's test function:

```javascript
/**
 * @typedef Context
 * @property {number} foo
 */
const test = /** @type {import('ava').TestFn<Context>} */ (require('ava').default);

test.before((t) => {
  t.context.foo = 42;
});
```

Typescript marks the above code as valid, and code suggestions work as expected.

## Actual behavior

ESLint as configured below, shows a warning for the line `t.context.foo = 42;`.
The warning is: `'@type' is redundant when using a type system      jsdoc/check-tag-names`

If you remove the `@type` declaration, then:
- ESLint is satisfied
- Typescript throws an error: `TS18046:  t.context  is of type  unknown`.

## ESLint Config

```json
{
  "extends": [
    "plugin:jsdoc/recommended-typescript"
  ]
}
```

## ESLint sample

See the example in `Expected behavior` above

## Environment

- Node version: v20.3.0
- ESLint version: v8.46.0
- `eslint-plugin-jsdoc` version: 46.4.6

**Comment by brettz9** [comment]
I think the idea is to use `as` type-assertions instead of a `@type` tag: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions

**Comment by Lalem001** [comment]
Am I using the wrong configuration?

**Comment by Lalem001** [comment]
Indeed. Thanks for the help.
